Add runtime api's and permissions to docs for extensibility team (#10915)
mendix/docs: content/en/docs/apidocs-mxsdk/apidocs/studio-pro-11/extensibility-api/web/web-extensions-howtos/permissions.md

---
title: "Extension Permissions in Overview Pane"
linktitle: "Extension Permissions"
url: /apidocs-mxsdk/apidocs/web-extensibility-api-11/extension-permissions/
---

## Introduction

This how-to describes how the permission system works for web extensions in Studio Pro. Permissions allow extensions to request access to sensitive APIs or data that require explicit user consent.

> 

Reviewer: ```suggestion

{{% alert color="info" %}}
Extension permissions were introduced in version 11.9.0.
{{% /alert %}}

```

## How Permissions Work

> Web extensions can request permissions to access sensitive functionalities. The permission system follows these principles:

Reviewer: ```suggestion
Web extensions can request permissions to access sensitive functionality. The permission system follows these principles:
```

> * **Opt-in by default** — Extensions cannot access protected APIs unless they explicitly request the permission and the user grants it

Reviewer: This looks wrong. It seems that the default is "Opt-out by default". I.e. you don't get access unless you specifically opt in by setting the permission to `true`?

I'd also avoid "user" and say "you grant it" assuming the reader is the person developing the app in Studio Pro.

Reviewer 2: @SofiBili Can you confirm the default setting?

> * **User control** — Users decide which permissions to grant through the Extensions Overview pane in Studio Pro

Reviewer: I guess this user is the developer of the app? Perhaps it would be clearer to say "developer" or "maker" than 
"user"? Or perhaps

```suggestion
* **User control** — You decide which permissions to grant through the Extensions Overview pane in Studio Pro
```

> * **Per-project settings** — Permission grants are stored per project so users can have different permission settings for the same extension across different projects

Reviewer: I find this difficult to understand. Are we just saying that each app has its own permission settings? then perhaps we should say something like:

```suggestion
* **Per-app settings** — Permission grants are stored in the app so you can set different permissions for the same extension depending on the app requirements
```

## Requesting Permissions

To request a permission, declare it in your extension's `package.json` file under the `mendix.permissions` object:

```json
{
  "mendixComponent": {
    "entryPoints": {
      "main": "main.js",
      "ui": {
        "tab": "tab.js"
      }
    },
    "permissions": {
      "runtime-configuration-private": true
    }
  }
}
```

> Setting a permission to **true** indicates that your extension requests this permission. The user must grant it before your extension can use the protected functionality.

Reviewer: Does setting it to false have any meaning? Perhaps we need to describe it anyhow.

Reviewer 2: @SofiBili Does setting a permission to false have a specific meaning?

Author: no, false is the default state so the code will ignore it

## Granting Permissions (User Flow)

When a user installs an extension that requests permissions, they can manage those permissions through the Extensions Overview pane. Follow the steps below:

> 1. Open Studio Pro and load a project with the extension installed.

Reviewer: Why are we using **project** here? The UX is **Open App…**

<img width="786" height="354" alt="Image" src="https://github.com/user-attachments/assets/e1a80f20-7c2e-4ee7-9ca0-a8db7d764fb6" />

Reviewer 2: @SofiBili Is this how a user loads a project, or is there a different location they are loaded? We haven't built one in the past, so we don't know the exact sets.

If they go from **File** > **Open App**, then we should use "app" instead of project. If there is another way extensions are loaded, then let's use the UI they follow there (which may be project?).

Author: yes app is the correct word, thank you!

2. Go to **View** > **Extensions Overview** to open the Extensions Overview pane.
3. Find the extension in the list. Under the extension details, the **Permissions** section displays the requested permissions.
4. Check or uncheck the checkbox next to each permission to grant or revoke access.

## Available Permissions

> Currently, the following permissions are available for web extensions:

Reviewer: ```suggestion
The following permissions are available for web extensions:
```
